### PR TITLE
finish card main

### PR DESCRIPTION
**Pull request by DominikReszczynski**
The part with the main cards is done for card rotation. At the moment, I'm using a function that rotates the text, but in the future, there's nothing stopping me from replacing the text in the function with something else.

### CODE REVIEW

File: src/components/AppCard.tsx [hunk 1]
```diff
@@ -12,12 +12,30 @@ export const AppCard = () => {
...
+
+	const reverseText = (index: number) => {
+		const newTextArr = [...text];
+		newTextArr[index] = newTextArr[index].split('').reverse().join('');
+		setText(newTextArr);
+	};
+
+	return (
+		<div>
+			{text.map((item, index) => (
```
Comment (Sumick) [inline src/components/AppCard.tsx]: **suggestion:** Let's add Prettier to your project, so you'll have everything formatted nicely

File: src/components/AppCard.tsx [hunk 2]
```diff
@@ -12,12 +12,30 @@ export const AppCard = () => {
...
 	];
-	return textArr.map((item, index) => (
-		<div className={styles.card} key={index}>
-			<div className={styles.editConteiner}>
-				<img src={editIcon} alt='edit' />
-			</div>
-			<h4 className={styles.text}>{item}</h4>
+	const [text, setText] = useState(textArr);
+
+	const reverseText = (index: number) => {
```
Comment (Sumick) [inline src/components/AppCard.tsx]: **issue:** I don't understand why single AppCard holds an array of strings? Each card should handle both sides. Currently in your implementation it's impossible to differ which side is currently active

File: src/components/AppCard.tsx (hunk 1 above)
Comment (Sumick) [inline src/components/AppCard.tsx]: **issue:** Hm, I don't understand what is happening here - I assume that `AppCard` is a single Card component which should handle both sides (front, and back). Currently you're mapping over array of strings. So is it a list? Why? What are these values?

**Review by Sumick — CHANGES_REQUESTED** [review]
This is not covering current task scope, as it's missing a lot of details. In this task, you should create a component for displaying added `Card`. You can think of it as you should have 2 different components. One is for creating new card (which shows when clicking on Add button in header), and second is for displaying added card in the list. I suggest you to split these 2 components so you are clear what does what specifically. You don't have to focus on rotating card now, as it's gonna be in other task. For now, we should focus on creating reusable UI components which we will glue later using logic. 

To sum up - the flow which you should have in this current moment:
1. When you click on Add button in header, and then NewCard components shows up. NewCard should be a component for adding new card, and it should support both sides
2. After adding (you don't have to handle adding logic now) it should show Card component, so basically you need other component which is similar, but it's for different case
3. You should structure your data somehow, like:

```ts
interface Flashcard {
 front: string;
 back: string;
}
```
and then use it as props for your card components.

File: src/components/AppCard.tsx [hunk 3]
```diff
@@ -1,23 +1,133 @@
...
-	return textArr.map((item, index) => (
-		<div className={styles.card} key={index}>
-			<div className={styles.editConteiner}>
-				<img src={editIcon} alt='edit' />
-			</div>
-			<h4 className={styles.text}>{item}</h4>
-		</div>
-	));
+
+	const [cards, setCards] = useState<Flashcard[]>(textArr);
```
Comment (Sumick) [inline src/components/AppCard.tsx:40]: **praise:** This state is properly declared, but this should be in other place. In this case, `App` component preferably

File: src/components/AppCardEdit.tsx [hunk 4]
```diff
@@ -1,29 +1,7 @@
 import React from 'react';
 import style from './AppCardEdit.module.css';
-import binIcon from '../images/Kind=Delete.svg';
+
 //card
 export const AppCardEdit = () => {
-	return (
-		<>
-			<div className={style.cardConteinerNext}>
```
Comment (Sumick) [inline src/components/AppCardEdit.tsx:8]: **issue:** It wasn't problem with this component. you should leave it as it was - we should split components for smaller pieces

File: src/components/AppCard.tsx [hunk 5]
```diff
@@ -1,23 +1,133 @@
...
+
+	return (
+		<>
+			{editMode && (
+				<div>
+					{cards.map((card, index) => (
+						<button
+							className={styles.cardBtn}
+							key={index}
+							onClick={() => toggleCardSide(index)}
```
Comment (Sumick) [inline src/components/AppCard.tsx:60]: **issue:** It got really messy - we should split it into smaller components, and not keep everything in one place

**Review by Sumick — CHANGES_REQUESTED** [review]
This is not good approach, as now you have everything packed now in one component. It got really messy, and it will be almost impossible to maintain or extend logic within. I will sketch you specific component structure based on your `app.tsx`, try to recreate it in your app. You will get the approach really fast:)

```tsx

import React, { useState } from 'react';
import AppLayout from './AppLayout';
import AppHeader from './AppHeader';
import AppCard from './AppCard';
import CreateNewCard from './AppCard';

function App() {
    const [cards, setCards] = useState([]);
    const [isAddingCard, setAdding] = useState(false);

    return (
        <AppLayout>
            <AppHeader cardsAmount={cards.length} onAddCard={() => setAdding(true)} />
            <div className='content'>
                {isAddingCard && <CreateNewCard />}
                <AppCard front="Hejka" back="Hey" />
            </div>
        </AppLayout>
    );
}

export default App;


```
